src/modules/genre.js

```diff
@@ -0,0 +1,151 @@
+
+// function display movies based on action genre
+const displayAction = () => {
+  getShows().then((data) => {
+    const shows = data.filter((show) => show.genres.includes('Action')).map((show) => `
+                  <div class="movie-card">
+                      <img src="${show.image.medium}" alt="${show.name}">
+                      <h3 class="movie-title">${show.name}</h3>
+                      <div class="interactions">
+                          <button class="comments">comment</button>
+                          <i class="like fa-regular fa-heart"></i>
+                      </div>
+                  </div>
+                  `).join('');
+    movieList.innerHTML = shows;
+  });
+};
+
+// function display movies based on drama genre
+const displayDrama = () => {
+  getShows().then((data) => {
+    const shows = data.filter((show) => show.genres.includes('Drama')).map((show) => `
+                  <div class="movie-card">
+                      <img src="${show.image.medium}" alt="${show.name}">
+                      <h3 class="movie-title">${show.name}</h3>
+                      <div class="interactions">
+                          <button class="comments">comment</button>
+                          <i class="like fa-regular fa-heart"></i>
+                      </div>
+                  </div>
+                  `).join('');
+    movieList.innerHTML = shows;
+  });
+};
+
+// function display movies based on comedy genre
+const displayComedy = () => {
+  getShows().then((data) => {
+    const shows = data.filter((show) => show.genres.includes('Comedy')).map((show) => `
+                  <div class="movie-card">
+                      <img src="${show.image.medium}" alt="${show.name}">
+                      <h3 class="movie-title">${show.name}</h3>
+                      <div class="interactions">
+                          <button class="comments">comment</button>
+                          <i class="like fa-regular fa-heart"></i>
+                      </div>
+                  </div>
+                  `).join('');
+    movieList.innerHTML = shows;
+  });
+};
+
+// function display movies based on romance genre
+const displayRomance = () => {
+  getShows().then((data) => {
+    const shows = data.filter((show) => show.genres.includes('Romance')).map((show) => `
+                      <div class="movie-card">
+                          <img src="${show.image.medium}" alt="${show.name}">
+                          <h3 class="movie-title">${show.name}</h3>
+                          <div class="interactions">
+                              <button class="comments">comment</button>
+                              <i class="like fa-regular fa-heart"></i>
+                          </div>
+                      </div>
+                      `).join('');
+    movieList.innerHTML = shows;
+  });
+};
+
+// function display all movies
+allBtn.addEventListener('click', () => {
```

Comment:
I think it would be better if we had a single function for these specific genres. The genre could be passed to the function as an argument like `displayGenre(genre) ....` then in the filter we can have `data.filter(show) => show.genres.includes(genre)`. This will help us implement the DRY technique. Imagine if we had say `1000` genres, then we'd need to write `1000` display functions, but this will help us have one function, irrespective of the number of genres we have.